### PR TITLE
fill some gaps wrt honoring disablement of the plugin

### PR DESCRIPTION
**Pull request by gabemontero**
fixes https://github.com/openshift/jenkins-sync-plugin/issues/168

@openshift/devex @scoheb fyi

### CODE REVIEW

**Comment by gabemontero** [comment]
[test]

File: src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java [hunk 1]
```diff
@@ -50,36 +50,35 @@
...
     @DataBoundConstructor
     @SuppressFBWarnings("EI_EXPOSE_REP2")
     public PipelineJobListener(String server, String[] namespaces) {
         this.server = server;
         this.namespaces = namespaces;
-        init();
-    }
-
-    private void init() {
-        namespaces = OpenShiftUtils.getNamespaceOrUseDefault(namespaces,
```
Comment (scoheb) [inline src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java]: is this going to cause problems that it is no longer called?

Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java]: yeah this was a consideration for me as well, but 1) namespaces is not used in this class at all, 2) `OpenShiftUtils.getNamespaceOrUseDefault` has no side effects, and 3) our regression tests passed 

so imo this is dead code

technically speaking the server instance variable is not used either ... I am also looking to removing the databoundconstructor (the object should still be instantiated given the ItemExtension listener).

now that the clean up in `init()` has passed, I'm looking to clean that up next.

Comment (gabemontero) [reply src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java]: yep confirmed locally that listener still comes up without the databoundconstructor ... pushing updates in a bit

**Comment by gabemontero** [comment]
docker device mapper hiccup during the build ... trying test again

**Comment by gabemontero** [comment]
[test]

**Comment by gabemontero** [comment]
[merge]

**Pull request merged**